adapters/src/main/java/com/squareup/moshi/adapters/ConfigurableEnumJsonAdapter.java

```diff
@@ -0,0 +1,72 @@
+package com.squareup.moshi.adapters;
+
+import com.squareup.moshi.Json;
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonDataException;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import javax.annotation.Nullable;
+
+/**
+ * A JsonAdapter for enums that allows having a fallback enum value when a deserialized string does
+ * not match any enum value.
+ */
+public final class ConfigurableEnumJsonAdapter<T extends Enum<T>> extends JsonAdapter<T> {
```

Comment:
I wonder if this should be just called EnumJsonAdapter ? The other one isn’t a public type so I don’t think it matters

Reply:
yes, totally. good call.